src/contexts/SettingProvider.jsx

```diff
@@ -0,0 +1,19 @@
+import PropTypes from "prop-types";
+
+import useSetting from "~/hooks/useSetting";
+
+import SettingContext from "./settingContext";
+
+const url = "28912ddd5410287eb4c0";
+const SettingProvider = ({ children }) => {
+  const { state, dispatch } = useSetting(url);
+
+  return state.setting.menu.length > 0 ? (
+    <SettingContext.Provider value={{ state, dispatch }}>{children}</SettingContext.Provider>
+  ) : null;
```

Comment:
Please check AuthContext file to follow,
basically no need to separate context and provider
and in context folder please use PascalCase naming and only have XyzContext namings